clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -519,6 +519,42 @@ public static String stackTrace(Throwable e) {
         return sw.toString();
     }
 
+    /**
+     * returns the root cause of a given throwable
+     * @param t a throwable
+     * @return the root cause of the argument
+     */
+    public static Throwable rootCause(Throwable t) {
+        Throwable cause = t;
+        while (cause != null && cause.getCause() != null && cause.getCause() != cause) {
+            cause = cause.getCause();
+        }
+        return cause;
+    }
+
+    /**
+     * returns a short human-readable description of a throwable
```

Comment:
It will be worth mentioning that it includes the root cause since this is in `Utils`.